torch/utils/data/datapipes/map/combining.py

```diff
@@ -0,0 +1,45 @@
+from torch.utils.data import MapDataPipe, functional_datapipe
+from typing import Optional, Sized, Tuple, TypeVar
+
+T_co = TypeVar('T_co', covariant=True)
+
+
+@functional_datapipe('concat')
+class ConcatMapDataPipe(MapDataPipe):
+    r""" :class:`ConcatMapDataPipe`.
+
+    Map DataPipe to concatenate multiple Map DataPipes.
+    The actual index of is the cumulative sum of source datapipes.
+    For example, if there are 2 source datapipes both with length 5,
+    index 0 to 4 of the resulting `ConcatMapDataPipe` would refer to
+    elements of the first datapipe, and 5 to 9 would refer to elements
+    of the second datapipe.
+    args:
+        datapipes: Map DataPipes being concatenated
+    """
+    datapipes: Tuple[MapDataPipe]
+    length: Optional[int]
+
+    def __init__(self, *datapipes: MapDataPipe):
+        if len(datapipes) == 0:
+            raise ValueError("Expected at least one DataPipe, but got nothing")
+        if not all(isinstance(dp, MapDataPipe) for dp in datapipes):
+            raise TypeError("Expected all inputs to be `MapDataPipe`")
+        if not all(isinstance(dp, Sized) for dp in datapipes):
+            raise TypeError("Expected all inputs to be `Sized`")
+        self.datapipes = datapipes  # type: ignore[assignment]
+        self.length = None
```

Comment:
To solve the lint error. You can switch length from optional[int] to int, since we ensured all prior datapipes have valid length

Reply:
Emmm. Sorry if I make it confusing. Since we change the annotation to `int`, you should calculate `length` here to assign `self.length` an int value.
Then `__len__` can directly return `self.length`